### PR TITLE
WIP: Supporting Ignore Collision tag in URDF

### PR DESCRIPTION
**Pull request by vidurvij-Unity**

### CODE REVIEW

File: Runtime/RosSharpDefinitions/Robot.cs [hunk 1]
```diff
@@ -92,6 +94,14 @@ private List<Plugin> ReadPlugins(XElement node)
             return plugins.ToList();
         }
 
+        private List<Tuple<string,string>> ReadDisableCollision(XElement node)
+        {
+            var disable_collisions =
```
Comment (peifeng-unity) [inline Runtime/RosSharpDefinitions/Robot.cs:99]: nit: I think we should use less "var" implicit types. we could use `System.Collections.Generic.IEnumerable<Tuple<string, string>>` as the explicit type

File: Runtime/UrdfComponents/UrdfRobot.cs [hunk 2]
```diff
@@ -79,6 +82,31 @@ public void SetAxis(ImportSettings.axisType setAxis)
...
         }
 
+        void Start()
+        {
+            CreateCollisionExceptions();
+        }
+
+        public void CreateCollisionExceptions()
+        {
+            int j = 0;
```
Comment (peifeng-unity) [inline Runtime/UrdfComponents/UrdfRobot.cs:92]: why do we need the variable `j`?

File: Runtime/UrdfComponents/UrdfRobot.cs [hunk 3]
```diff
@@ -79,6 +82,31 @@ public void SetAxis(ImportSettings.axisType setAxis)
...
+            if (collisionExceptions != null)
+            {
+                foreach (var ignoreCollision in collisionExceptions)
```
Comment (peifeng-unity) [inline Runtime/UrdfComponents/UrdfRobot.cs:95]: nit: I think we should use less implicit type `var`.

File: Editor/NeedsRuntimeConversion/Extensions/UrdfRobotExtensions.cs [hunk 4]
```diff
@@ -90,6 +91,24 @@ public static void CorrectAxis(GameObject robot, ImportSettings.axisType axis =
...
 
         }
+
+        private static void CreateCollisionExceptions(Robot robot, GameObject robotGameObject)
+        {
+            string collisionObjectName = "Collisions";
+            var ColliisonList = new List<CollisionIgnore>();
+            if (robot.ignoreCollisionPair.Count > 0)
+            {
+                foreach (var ignoreCollision in robot.ignoreCollisionPair)
```
Comment (peifeng-unity) [inline Editor/NeedsRuntimeConversion/Extensions/UrdfRobotExtensions.cs:101]: nit: same, I think we should not use implicit type `var` if possible

File: Editor/NeedsRuntimeConversion/Extensions/UrdfRobotExtensions.cs [hunk 5]
```diff
@@ -90,6 +91,24 @@ public static void CorrectAxis(GameObject robot, ImportSettings.axisType axis =
...
+                {
+                    Transform colliisonObject1 = GameObject.Find(ignoreCollision.Item1).transform.Find(collisionObjectName);
```
[the remark this answers is not in the document]
Comment (vidurvij-Unity) [reply Editor/NeedsRuntimeConversion/Extensions/UrdfRobotExtensions.cs:103]: This is taken care of when we are parsing the file. It won't load 2 links with the same name.

File: Runtime/RosSharpDefinitions/Robot.cs (hunk 1 above)
Comment (vidurvij-Unity) [reply Runtime/RosSharpDefinitions/Robot.cs:99]: This was in line with code written previously. This code was part of ROS #

File: Editor/NeedsRuntimeConversion/Extensions/UrdfRobotExtensions.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (peifeng-unity) [reply Editor/NeedsRuntimeConversion/Extensions/UrdfRobotExtensions.cs:103]: Okay; does URDF requires different names on different links? I didn't find the requirements in the doc (http://wiki.ros.org/urdf/XML/link)

**Review by peifeng-unity — APPROVED** [review]

**Pull request merged**